server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -873,16 +872,14 @@ public Response isHandOffComplete(
       final DateTime now = DateTimes.nowUtc();
 
       // A segment that is not eligible for load will never be handed off
-      boolean notEligibleForLoad = true;
+      boolean eligibleForLoad = false;
       for (Rule rule : rules) {
         if (rule.appliesTo(theInterval, now)) {
-          if (rule instanceof LoadRule) {
-            notEligibleForLoad = false;
-          }
+          eligibleForLoad = rule.shouldSegmentBeLoaded();
           break;
```

Comment:
I don't think we should break anymore if there are multiple rules that apply. 
Suppose the first rule is 0-replica rule, and the second is a 2-replica rule for a different tier, this implementation might break saying that `eligibleForLoad = false`, but it should have been true and waited for handoff

Reply:
rules are applied in sequence so if the first rule is a 0-replica rule, wouldn't coordinator just stop there? A 0-replica rule must come after a 2-replica rule. that's how most load/drop rules work unless something recently changed.

Reply:
Apologies, I had assumed wrongly that the check was on tier-wise basis, but a single rule has the replication factor for all the tiers.